notebooks/concepts/specviz_glue_unit_conversion.ipynb

```diff
@@ -0,0 +1,293 @@
+    "from jdaviz import Specviz"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0d253733",
+   "metadata": {},
+   "source": [
+    "First spectrum."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cba69bb7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wave1 = np.linspace(2, 5, 10) * u.um\n",
+    "flux1 = [1, 2, 3, 4, 5, 5, 4, 3, 2, 1] * u.Jy\n",
+    "spec1 = Spectrum1D(flux=flux1, spectral_axis=wave1)\n",
+    "\n",
+    "print(wave1)\n",
+    "print(flux1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1e0e7dc5",
+   "metadata": {},
+   "source": [
+    "Second spectrum in different units and with slight offsets in spectral axis."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "26bc794d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wave2 = (wave1 + (0.1 * u.um)).to(u.GHz, u.spectral())\n",
+    "flux2 = flux1.to(u.mJy)\n",
+    "spec2 = Spectrum1D(flux=flux2, spectral_axis=wave2)\n",
+    "\n",
+    "print(wave2)\n",
+    "print(wave2.to(u.um, u.spectral()))\n",
+    "print(flux2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2bb4f4ea",
+   "metadata": {},
+   "source": [
+    "Fire up Specviz."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "74692a02",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "specviz = Specviz()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "efdba701",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "specviz.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "642fbae1",
+   "metadata": {},
+   "source": [
+    "Load the data into Specviz. Desired behavior:\n",
+    "\n",
+    "1. \"Jy_um\" would load with Jy in Y-axis and um in X-axis.\n",
+    "2. \"mJy_GHz\" would load with data automatically converted to Jy and um in the plot. You would see the same shape but slightly offset in X-axis, just slightly."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5ee09160",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "specviz.load_data(spec1, data_label=\"Jy_um\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9655292b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "specviz.load_data(spec2, data_label=\"mJy_GHz\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e6012551",
+   "metadata": {},
+   "source": [
+    "Change the spectral axis display unit to GHz."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6a4611b3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "viewer = specviz.app.get_viewer(\"spectrum-viewer\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ac319e71",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "viewer.state.x_display_unit = \"GHz\""
```

Comment:
This fails now, seems like the notebook should refer to Hz instead of GHz:
```
File ~/projects/jdaviz/jdaviz/core/template_mixin.py:674, in UnitSelectPluginComponent._selected_changed(self, event)
    672 if new_u not in self.unit_choices:
    673     self.selected = event['old']
--> 674     raise ValueError(f"{event['new']} not one of {self.labels}, reverting selection to {event['old']}")  # noqa
    676 # convert to default string representation from the valid choices
    677 ind = self.unit_choices.index(new_u)

ValueError: GHz not one of ['Angstrom', 'Hz', 'erg', 'nm', 'um', 'Ci', 'J', 'Ry', 'cm', 'eV', 'k', 'm'], reverting selection to Hz
```

Reply:
The string compare stuff is annoying like that, you have to include all the possible prefixes or it is going to fail.

I wonder if @astrofrog can make it less annoying but if not, we need to use the "include prefix" option in some of these base units.

Reply:
I pushed a commit which now allows setting _any_ unit accepted by glue via the API.  If that unit does not appear in the list of dropdown choices, it will be appended to the end of the available list for the remainder of the session.

For example:

```
uc = specviz.plugins['Unit Conversion']
uc.open_in_tray()
print(uc.spectral_unit.choices)  #  ['Angstrom', 'Hz', 'erg', 'nm', 'um', 'Ci', 'J', 'Ry', 'cm', 'eV', 'k', 'm']
uc.spectral_unit = 'GHz'
print(uc.spectral_unit.selected)  # GHz
print(uc.spectra_unit.choices)  # ['Angstrom', 'Hz', 'erg', 'nm', 'um', 'Ci', 'J', 'Ry', 'cm', 'eV', 'k', 'm', 'GHz']
```